backend/src/main/java/pl/edu/agh/controller/MatchManagementController.java

```diff
@@ -30,4 +30,38 @@ public ResponseEntity<List<Match>> getMatches() {
     public ResponseEntity<Map<RefereeType, UUID>> addNewMatch(@RequestBody Match match) {
         return new ResponseEntity<>(matchManagementService.addNewMatch(match), HttpStatus.OK);
     }
+
+    @GetMapping("/match-by-token/{token}")
+    public Match getMatchByToken(@PathVariable String token) {
```

Comment:
To be consistent, we should return ResponseEntity with the appropriate status code (as it was done in other methods) instead of Match object itself